### PR TITLE
WebSocket RPC

### PR DESCRIPTION
**Pull request by thundering-silence**
# Task: RYSK-259

## Description

Implement web-socket based event listener for updates to chainlink's aggregator instead of polling it every 6 seconds.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update
- [ ] Quality of life improvement
- [ ] Package upgrades

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have NATSPEC'd any new functions
- [ ] If appropriate, I have properly tested my new functionality

### CODE REVIEW

File: packages/front-end/src/hooks/useEthUsdPriceChangeHandler.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    state: {
+      options: { refresh },
+    },
+  } = useGlobalContext();
+
+  const { data: aggregatorAddress } = useContractRead({
+    abi: EACAggregatorProxyABI,
+    functionName: "aggregator",
+    address: ETH_USD_AGGREGATOR_ADDRESS,
+  });
```
Comment (MeanBoyCousin) [inline packages/front-end/src/hooks/useEthUsdPriceChangeHandler.ts:20]: Does the address returned from here change then? Just wondering why we need to call a contract to get an address.

**Review by MeanBoyCousin — COMMENTED** [review]
Couple of things to modify. The other issue is the new ABI files. If you check the contracts package, you will find a file called `copyABIsToFrontend.sh` which runs an `rm -rf` on the ABIs folder when new contracts are compiled. Admittedly this isn't often but could lead to an easy/silly mistake. I'd advise you to place these files in JSON format inside `packages\contracts\abis` and then amend the shell script (line 54) to include them. This will auto-generate them in the FE when you run `yarn workspace contracts compile`.

File: packages/front-end/src/hooks/useEthUsdPriceChangeHandler.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  useContractEvent({
+    address: aggregatorAddress,
+    abi: AggregatorV3ABI,
+    eventName: "AnswerUpdated",
+    listener: (x, y, updatedAt) => {
```
[the remark this answers is not in the document]
Comment (thundering-silence) [reply packages/front-end/src/hooks/useEthUsdPriceChangeHandler.ts:26]: yeah I couldn't use `_` for both so resorted to `x` and `y` - replaced with `_` and `__` 👍

File: packages/front-end/src/hooks/useEthUsdPriceChangeHandler.ts (hunk 1 above)
Comment (thundering-silence) [reply packages/front-end/src/hooks/useEthUsdPriceChangeHandler.ts:20]: The contract at `ETH_USD_AGGREGATOR_ADDRESS` is a proxy to another contract that is the one actually emitting events (we listen to) and storing the price. Chainlink can freely change the underlying contract so we might end up listening for events on a legacy contract. That is why I implemented this call first

**Review by MeanBoyCousin — APPROVED** [review]
Good stuff. Just needs the merges from the trunk removed and we are good.

**Review by MeanBoyCousin — APPROVED** [review]

**Pull request merged**